### DISCUSSION TITLE
"Argument 'foo' was not set" after update to 4.1 in Cake Frosting

### DISCUSSION THREAD

**Discussion by pascalberger** [cake-build/cake#4373] [General]
Having the following code in a Cake Frosting project:

```
var value = context.Argument<string>("foo");
```

When running Cake with `--foo=bar` it worked in 4.0, but with 4.1 an `Argument 'foo' was not set` error is thrown.

See https://github.com/pascalberger/cake-4373 for a repro.

**Comment by pascalberger** [comment]
Workaround:

```
var value = context.Argument<string>("--foo");
```

**Reply by devlead** [reply]
Yip, but this is a bug.

**Reply by devlead** [reply]
Won't work in 4.2.

**Comment by pascalberger** [comment]
Issue has been fixed in 4.2 which is now available
